dashboards-observability/public/components/notebooks/components/create.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/* eslint-disable react-hooks/exhaustive-deps */
+import {
+    EuiButton,
+    EuiFieldText,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiForm,
+    EuiFormRow,
+    EuiHorizontalRule,
+    EuiPage,
+    EuiPageBody,
+    EuiPageContent,
+    EuiPageContentHeader,
+    EuiPageContentHeaderSection,
+    EuiPageHeader,
+    EuiPageHeaderSection,
+    EuiSpacer,
+    EuiTitle,
+    EuiToolTip,
+  } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import "../../../../public/components/application_analytics/app_analytics.scss"
+import { ChromeBreadcrumb } from '../../../../../../src/core/public';
+
+  interface CreateNotebookProps {
+    existingNotebookId: string;
+    parentBreadcrumb: ChromeBreadcrumb;
+    setBreadcrumbs: (newBreadcrumbs: ChromeBreadcrumb[]) => void;
+    createNotebook: (name: string) => void;
+    renameNotebook: (name: string, id: string) => void
+  }
+  
+  export const CreateNotebook = (props: CreateNotebookProps) => {
+    const {
+        existingNotebookId,
+        parentBreadcrumb,
+        setBreadcrumbs,
+        createNotebook,
+        renameNotebook,
+    } = props;
+  
+    const editMode = existingNotebookId !== 'undefined';
+
+    const [name, setName] = useState(sessionStorage.getItem('NotebooksName') || '');
+    const isDisabled = !name;
+
+    const onCreate = (name: string) => {
+      createNotebook(name);
+      sessionStorage.setItem('NotebooksName', '');
```

Comment:
Is there a hard requirement to use `sessionStorage` here? Side Question: Why are we resetting `NotebooksName` in sessionStorage after creation?

Reply:
No hard requirement, but again wanted to keep some parity between application analytics and notebooks. I think only reason is to not lose page content on refresh, but not sure how valuable that is. The reset is so that if they create something and then try to create another one they will not see their old notebook name when trying to create a new one.